llm/src/tokenizer.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2022 MosaicML Benchmarks authors
+# SPDX-License-Identifier: Apache-2.0
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+import transformers
+
+
+class LLMTokenizer(ABC):
+    tokenizer_name: str
+    max_seq_len: int
+
+    def __init__(self, tokenizer_name: str, max_seq_len: int):
+        self.tokenizer_name = tokenizer_name
+        self.max_seq_len = max_seq_len
+
+    @abstractmethod
+    def __call__(self, *args, **kwargs):
+        raise NotImplementedError
+
+    @abstractmethod
+    def encode(self, x):
+        raise NotImplementedError
+
+    def decode(self, x):
```

Comment:
if `def encode(self, x):` has the `@abstractmethod` decorator, should `def decode(self, x):` also have the same decorator?